provider/agent_test.go

```diff
@@ -211,6 +211,98 @@ func TestAgent_Metadata(t *testing.T) {
 	})
 }
 
+func TestAgent_ResourcesMonitoring(t *testing.T) {
+	t.Parallel()
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: coderFactory(),
+		IsUnitTest:        true,
+		Steps: []resource.TestStep{{
+			Config: `
+				provider "coder" {
+					url = "https://example.com"
+				}
+				resource "coder_agent" "dev" {
+					os = "linux"
+					arch = "amd64"
+					resources_monitoring {
+						memory {
+							enabled = true
+							threshold = 80
+						}
+						volume {
+							path = "/volume1"
+							enabled = true
+							threshold = 80
+						}
+						volume {
+							path = "/volume2"
+							enabled = true
+							threshold = 100
+						}
+					}
+				}`,
+			Check: func(state *terraform.State) error {
+				require.Len(t, state.Modules, 1)
+				require.Len(t, state.Modules[0].Resources, 1)
+
+				resource := state.Modules[0].Resources["coder_agent.dev"]
+				require.NotNil(t, resource)
+
+				t.Logf("resource: %v", resource.Primary.Attributes)
+
+				attr := resource.Primary.Attributes
+				require.Equal(t, "1", attr["resources_monitoring.#"])
+				require.Equal(t, "1", attr["resources_monitoring.0.memory.#"])
+				require.Equal(t, "2", attr["resources_monitoring.0.volume.#"])
+				require.Equal(t, "80", attr["resources_monitoring.0.memory.0.threshold"])
+				require.Equal(t, "/volume1", attr["resources_monitoring.0.volume.0.path"])
+				require.Equal(t, "100", attr["resources_monitoring.0.volume.1.threshold"])
+				require.Equal(t, "/volume2", attr["resources_monitoring.0.volume.1.path"])
+				return nil
+			},
+		}},
+	})
+}
+
+func TestAgent_ResourcesMonitoring_OnlyMemory(t *testing.T) {
```

Comment:
Please add tests for unhappy cases like:
- multiple memory blocks
- volume blocks without paths defined
- thresholds outside accepted ranges
- ...etc

Reply:
I added some test cases which should improve a lot the coverage : 
- duplicate path
- multiple memory blocks
- thresholds out of range ( 0 - 100 )
and some OK tests.